app-api/models/datasets_and_traces.py

```diff
@@ -217,6 +217,8 @@ class DatasetPolicy(BaseModel):
     enabled: bool
     # the mode of this policy (e.g. block, log, etc.)
     action: str
+    # extra metadata for the policy (can be used to store internal extra data about a guardrail)
+    extra_metadata: dict = Field(default_factory=dict)
```

Comment:
allows us to store extra metadata in deployed guardrails (e.g. job ID they are coming from)